<launch>

  <arg name="name" default="warty"/>

  <!-- Settings for ATAK communications -->
  <arg name="callsign" default="Warty01" 
    doc="The callsign used by the robot on the ATAK system"/>
  <arg name="team_name" default="Ground Robots"  
    doc="The Team Name used by the robot on the ATAK system"/>
  <arg name="team_role" default="Autonomous Agent"  
    doc="The Team Role used by the robot on the ATAK system"/>
  <arg name="tak_ip" default="10.13.0.10"  
    doc="The IP address of the ATAK server"/>
  <arg name="tak_port" default="8088"  
    doc="The Port number on the ATAK server to connect to"/>

  <!-- Default location is River Courts, WPNY -->
  <arg name="start_latitude" default="41.39058" 
    doc="The initial latitude when the robot's odom is all zeros."/> 
  <arg name="start_longitude" default="-73.95323"
    doc="The initial longitude when the robot's odom is all zeros."/> 
  <arg name="start_altitude" default="2.0"
    doc="The initial altitude when the robot's odom is all zeros."/>
  <arg name="start_heading" default="0.0"/> <!-- A heading of 0 indicates due west -->

  <group ns="$(arg name)/atak">

    <node name="dcist_cots" pkg="atak_bridge" type="dcist_cots.py" output="screen">
        <param name="callsign" value="$(arg callsign)" />
        <param name="team_name" value="$(arg team_name)" />
        <param name="team_role" value="$(arg team_role)" />
        <param name="tak_ip" value="$(arg tak_ip)" />
        <param name="tak_port" value="$(arg tak_port)" />
    </node>

    <node name="global_cords" pkg="atak_bridge" type="global_cords_node.py" output="screen">
        <param name="name" value="$(arg name)" />
        <param name="start_latitude" type="double" value="$(arg start_latitude)" />
        <param name="start_longitude" type="double" value="$(arg start_longitude)" />
        <param name="start_altitude" type="double" value="$(arg start_altitude)" />
        <param name="start_heading" type="double" value="$(arg start_heading)" />
    </node>

    <node name="goto_target" pkg="atak_bridge" type="goto_target.py" output="screen">
        <param name="name" value="$(arg name)" />
        <param name="start_latitude" type="double" value="$(arg start_latitude)" />
        <param name="start_longitude" type="double" value="$(arg start_longitude)" />
        <param name="start_altitude" type="double" value="$(arg start_altitude)" />
        <param name="start_heading" type="double" value="$(arg start_heading)" />
    </node>

  </group> 
  
  


</launch>
